packages/flutter_tools/lib/src/ios/simulators.dart

```diff
@@ -700,12 +700,12 @@ class _IOSSimulatorLogReader extends DeviceLogReader {
   String _filterDeviceLine(String string) {
     final Match match = _mapRegex.matchAsPrefix(string);
     if (match != null) {
-      final String category = match.group(2);
-      final String tag = match.group(3);
-      final String content = match.group(4);
+      final String category = match.group(1);
+      final String tag = match.group(2);
+      final String content = match.group(3);
 
       // Filter out non-Flutter originated noise from the engine.
-      if (_appName != null && category != _appName) {
+      if (_appName != null && category != _appName && !category.endsWith(' $_appName')) {
```

Comment:
2. The last attempt at this PR was reverted because of a g3 test that checked the format of a simulator log from some older iOS SDK.  I added that test at https://github.com/flutter/flutter/pull/55808, and it passes in this PR.

This code is filtering out log lines that come from an app other than this one (category doesn't match the app name).  The previous regex assumed the app name was one word (Runner), but now we don't know how many words it will be, and it's hard to separate it out from the hostname, plus we don't know how many tokens the date will have since it's different formats across iOS SDKs.  So now it checks if anything after the date (green match below) ENDS in the app name instead of exactly matches.

<img width="1195" alt="Screen Shot 2020-04-30 at 10 29 36 AM" src="https://user-images.githubusercontent.com/682784/80740810-80ad6f00-8acd-11ea-9556-adacd146f8a8.png">

Reply:
This is a great comment, and the screenshot helps as well. That being said, I was completely bewildered from just reading the code. Do you think you can add more context to the comments here (especially why you're using `.endsWith()`?

Reply:
`category != _appName && !category.endsWith(' $_appName')` is redundant.